### PR TITLE
[Navigation] Add support for "floating" secondaryActions that appear on hover

### PR DESCRIPTION
**Pull request by henryyi**
### WHY are these changes introduced?

Resolves: Shopify/core-workflows/issues/634

This adds support for actions to "float" above the text in the navigation item. This is in line with the new design for app and channel navigation items in which the "action icons" appear to the left of the badge and they can be hidden and appear on hover. The actions overlay the text with a slight fade.
It was necessary to implement this feature in the Polaris lib to fix a few accessibility issues since features were previous implemented external to the Navigation.Item so things such as tab order were not correct.

The outcome of this change should not affect the current admin navigation since the "display on hover" behaviour already exists.

### WHAT is this pull request doing?

- secondarionActions moved to the left of the badge
- if secondarionActions the badge cannot reside inside the anchor and so is no longer clickable to go to the item link.
- a new prop `displayActionsOnHover` was added to enable the floating actions
- styling was updated to acccomodate the new feature

### How to 🎩

Can be tested in [Polaris storybook](https://polaris.hnav.jita-yi.us.spin.dev/?path=/story/all-components-navigation--with-multiple-secondary-actions-for-an-item)

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by chloerice — APPROVED** [review]
Just need to add the example and prop documentation for this to the website as well by adding a file [here](https://github.com/Shopify/polaris/tree/main/polaris.shopify.com/pages/examples)

**Comment by henryyi** [comment]
> Just need to add the example and prop documentation for this to the website as well by adding a file [here](https://github.com/Shopify/polaris/tree/main/polaris.shopify.com/pages/examples)

Thanks. I will create the examples in a followup PR.

File: polaris-react/src/components/Navigation/components/Item/Item.tsx [hunk 1]
```diff
@@ -241,9 +241,21 @@ export function Item({
...
+  const itemContentMarkup = (
+    <>
+      {iconMarkup}
+      {labelMarkup}
+      {secondaryActionMarkup ? null : wrappedBadgeMarkup}
+    </>
+  );
+
+  const outerContentMarkup = (
+    <>{secondaryActionMarkup ? wrappedBadgeMarkup : null}</>
```
Comment (kushp) [inline polaris-react/src/components/Navigation/components/Item/Item.tsx:256]: nit: this doesn't need to be inside <>

**Review by kushp — APPROVED** [review]
LGTM

File: polaris-react/src/components/Navigation/components/Item/Item.tsx (hunk 1 above)
Comment (henryyi) [reply polaris-react/src/components/Navigation/components/Item/Item.tsx:256]: I think I had more here before but good catch.

File: polaris-react/src/components/Navigation/Navigation.scss [hunk 2]
```diff
@@ -238,7 +246,7 @@ $disabled-fade: 0.6;
   }
 }
 
-.ItemInnerWrapper-Selected {
+.ItemInnerWrapper-selected {
```
Comment (Rmnlly) [inline polaris-react/src/components/Navigation/Navigation.scss:249]: Oh good change!

File: polaris-react/src/components/Navigation/components/Item/Item.tsx [hunk 3]
```diff
@@ -335,28 +368,25 @@ export function Item({
...
+              {itemLinkMarkup}
+              {secondaryActionMarkup}
+            </span>
+          ) : (
+            <>
+              {itemLinkMarkup}
+              {secondaryActionMarkup}
+            </>
+          )}
+          {outerContentMarkup}
```
Comment (Rmnlly) [inline polaris-react/src/components/Navigation/components/Item/Item.tsx:389]: Notice-ish: We may end up having a conflict here as my hover PR changed this considerably to remove the ternary from the render and into the itemLinkMarkup/another method.

**Review by Rmnlly — APPROVED** [review]
🎩  LGTM

**Pull request merged**